File: Front_GestionTempsDeTravail/src/app/entity/temps.ts
import { Time } from "@angular/common";

export class Temps {
  idTemps?: number; // Make idTemps optional by using '?'
  idEmployee: number | undefined;
  date: Date | undefined;
  heureDebut: String | undefined;
  heureFin: String | undefined;
  idPause: number | undefined;
}

File: Back_GestionTempsDeTravail/src/main/java/com/employee/back_gestionTempsDeTravail/entity/Temps.java
package com.employee.back_gestionTempsDeTravail.entity;

import java.sql.Date;
import java.sql.Time;
import java.util.List;

import jakarta.persistence.*;
import lombok.*;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Temps {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idTemps;

    private Date date;
    private String heureDebut;
    private String heureFin;

    @ManyToOne
    @JoinColumn(name = "idEmployee")
    private Employee employee;

    @OneToMany(mappedBy = "temps")
    private List<Pauses> pauses;

}
